.github/workflows/general.yaml

```
@@ -0,0 +1,59 @@
+name: "general"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```

```suggestion
  pull_request: {}
```

We want to build pull requests to any branch ideally